chapters/chap03-sec02.tex

```diff
@@ -18,28 +18,33 @@ \subsection{\mimc{} Encryption}\label{instantiation:mkhash:mimc-encryption}
     \caption{\mimc{} Encryption function.}\label{instantiation:fig:mimc}
 \end{figure*}
 
-\mimcEnc{} can be defined on both binary and prime fields. As such, the $\mathsf{OP}$ operation corresponds to either $\oplus$ or $+ \pmod{p}$~\cite{albrecht2016mimc, grassi2016mpc}. We denote by $\mimcPrime{p}$ (resp.~$\mimc{}_{2^n}$) the \mimcEnc{} function defined over \FFx{p} (resp.~\FFx{2^n}). Since block ciphers are usually defined over the product space of keys and messages, we consider the variables $c$, \rounds{} and \exponent{} as fixed. The functions' signatures thus become,
+\mimcEnc{} can be defined on both binary and prime fields. As such, the $\mathsf{OP}$ operation corresponds to either $\oplus$ or $+ \pmod{p}$~\cite{albrecht2016mimc, grassi2016mpc}. We denote by $\mimcPrime{p}$ (resp.~$\mimc{}_{2^n}$) the \mimcEnc{} function defined over \FFx{p} (resp.~\FFx{2^n}). Since block ciphers are usually defined over the product space of keys and messages, we consider the variables $c$, \rounds{} and \exponent{} as fixed. The function signatures thus become,
 \begin{align*}
     \mimcPrime{p} &: \FFx{p} \times \FFx{p} \to \FFx{p} \\
     \mimc{}_{2^n} &: \FFx{2^n} \times \FFx{2^n} \to \FFx{2^n}
 \end{align*}
 
-We will from now on only consider \mimc{} defined over prime fields as we consider SNARKs based on arithmetic circuits defined over a finite field \FFx{p}, $p$ prime.
+%% TODO: work in terms of \rCURVE instead of p, since \rCURVE is now generalized
```

Comment:
Removed the TODO.  It was more a reminder to raise this, since we now have an unbound variable, but no issue sticking to $p$.  Added the note about $p > 2$.

Reply:
> since we now have an unbound variable

Thanks, I see. That's a good point. To keep the scope of this "unbound variable" as small as possible, then what about phrasing the sentence like:
`From now on, we only consider \mimc{} defined over finite fields $\FFx{p}$, for prime $p > 2$, such that $\FFx{p}$ is the prime field over which \zksnark~operates (i.e.~$p = \rCurve$).`
?

Reply:
And use only `F_rcurve` after this sentence

Reply:
> And use only F_rcurve after this sentence
So you are suggesting we switch `$p$` to `$\rCURVE$` everywhere after this sentence?
If so, I'd suggest that we just remove the $p$ so the sentence becomes something like:
```
From now on, we only consider \mimc{} defined over prime fields.  In particular, the field $\FFx{\rCURVE}$ over which \zksnark~operates.
```
Let me know if that's your intention and I'll make the change.

Reply:
Perfect! (your suggestion above)

Reply:
I've pushed the change. `$p$` is used only to define the notation in terms of any prime, and when quoting a general result.  Those felt like they should be kept generic, but all other place now refer to `\rCURVE`.